lib/remap-functions/src/to_syslog_facility.rs

```diff
@@ -0,0 +1,242 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ToSyslogFacility;
+
+impl Function for ToSyslogFacility {
+    fn identifier(&self) -> &'static str {
+        "to_syslog_facility"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Integer(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(ToSyslogFacilityFn { value }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ToSyslogFacilityFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for ToSyslogFacilityFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let value = self.value.execute(state, object)?.try_integer()?;
+
+        // Facility codes: https://en.wikipedia.org/wiki/Syslog#Facility
+        let code = match value {
+            0 => Ok("kern"),
+            1 => Ok("user"),
+            2 => Ok("mail"),
+            3 => Ok("daemon"),
+            4 => Ok("auth"),
+            5 => Ok("syslog"),
+            6 => Ok("lpr"),
+            7 => Ok("news"),
+            8 => Ok("uucp"),
+            9 => Ok("cron"),
+            10 => Ok("authpriv"),
+            11 => Ok("ftp"),
+            12 => Ok("ntp"),
+            13 => Ok("security"),
+            14 => Ok("console"),
+            15 => Ok("solaris-cron"),
+            16 => Ok("local0"),
+            17 => Ok("local1"),
+            18 => Ok("local2"),
+            19 => Ok("local3"),
+            20 => Ok("local4"),
+            21 => Ok("local5"),
+            22 => Ok("local6"),
+            23 => Ok("local7"),
+            _ => Err(Error::from(format!("facility code {} not valid", value))),
+        };
+
+        match code {
+            Ok(code) => Ok(Value::from(code)),
+            Err(e) => Err(e),
+        }
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        use value::Kind;
+
+        self.value
+            .type_def(state)
+            .fallible_unless(Kind::Integer)
```

Comment:
```suggestion
            .is_fallible(true)
```

Unfortunately, because the fallibility depends on the integer value passed in, we can't know if this function will succeed at compile-time, and thus we'll have to consider it to always be fallible.

Reply:
I believe the call should be `into_fallible(true)` rather than `is_fallible(true)`.

`is_fallible()` just returns the value of `fallible`.